meta-balena-common/recipes-support/resin-init/resin-init-flasher/resin-init-flasher

```diff
@@ -192,39 +192,35 @@ IMAGE_FILE_SIZE=$(wc -c /opt/"$BALENA_IMAGE" | awk '{print $1}')
 
 # Check for UEFI mode and set variables accordingly.
 if [ -d /sys/firmware/efi ]; then
-    SECUREBOOT_VAR=$(efivar -p -n 8be4df61-93ca-11d2-aa0d-00e098032b8c-SecureBoot | awk 'NR==1, $1 == "Value:" {next}; NF {print $2}')
-    if [ "$SECUREBOOT_VAR" -eq "1" ]; then
+    SECUREBOOT_VAR=$(efivar -p -n 8be4df61-93ca-11d2-aa0d-00e098032b8c-SecureBoot \
+        | awk 'NR==1, $1 == "Value:" {next}; NF {print $2}')
+    SETUPMODE_VAR=$(efivar -p -n 8be4df61-93ca-11d2-aa0d-00e098032b8c-SetupMode \
+        | awk 'NR==1, $1 == "Value:" {next}; NF {print $2}')
+
+    if [ "$SECUREBOOT_VAR" = "1" ]; then
         LUKS=1
-    fi
-    SETUPMODE_VAR=$(efivar -p -n 8be4df61-93ca-11d2-aa0d-00e098032b8c-SetupMode | awk 'NR==1, $1 == "Value:" {next}; NF {print $2}')
-    if [ "$SETUPMODE_VAR" -eq "1" ]; then
-        inform "Secure boot setup mode detected - programming keys"
-        if ! mount | grep -q "type efivarfs"; then
-            mount -t efivarfs efivarfs "${EFIVARS_MOUNTDIR}"
-        fi
```

Comment:
This mount does nothing, it's inside a block guarded by a variable that will never be set unless the mount already exists.

Reply:
You're right, it's redundant